libobs/audio-monitoring/osx/coreaudio-output.c

```diff
@@ -54,6 +54,13 @@ static inline bool fill_buffer(struct audio_monitor *monitor)
 	return true;
 }
 
+static void on_pause_audio(void *data, calldata_t *calldata)
+{
+	UNUSED_PARAMETER(calldata);
```

Comment:
This should lock and release the monitor mutex as this is executed on the UI thread whereas the other calls happen either on the media IO thread or CoreAudio's own audio thread.

Reply:
Added mutex lock/unlock